Sources/Client/Utils/Atomic.swift

```diff
@@ -54,23 +66,10 @@ public final class Atomic<T> {
     
     /// Update the value safely.
     /// - Parameter changes: a block with changes. It should return a new value.
-    public func update(_ changes: @escaping (T) -> T?) {
-        queue.async(flags: .barrier) { [weak self] in
-            if let self = self, let oldValue = self.value {
-                self.updateValue(changes(oldValue))
-            }
-        }
-    }
-    
-    private func updateValue(_ newValue: T?) {
-        let oldValue = value
-        value = newValue
-        
-        if let callbackQueue = callbackQueue {
-            callbackQueue.async { [weak self] in self?.didSet?(newValue, oldValue) }
-        } else {
-            didSet?(newValue, oldValue)
-        }
+    public func update(_ changes: (T) -> T?) {
+        lock.lock()
+        value = changes(value!)
```

Comment:
Isn't it dangerous?

Reply:
YES! It's a leftover from my experiments. Thanks for finding it!